acceptancetests/assess_cross_model_relations.py

```diff
@@ -0,0 +1,264 @@
+        with second_bs_manager.existing_booted_context(
+                second_args.upload_tools):
+            second_model = second_bs_manager.client
+            ensure_cmr_offer_consumption(first_model, second_model)
+
+
+def ensure_cmr_offer_management(client):
+    """Ensure creation, listing and deletion of offers work.
+
+    Deploy dummy-source application onto `client` and offer it's endpoint.
+    Ensure that:
+      - The offer attempt is successful
+      - The offer is shown in 'list-offers'
+      - The offer can be deleted (and no longer appear in 'list-offers')
+
+    :param client:   ModelClient used to create a new model and attempt 'offer'
+      commands on
+    """
+    with temporary_model(client, 'offer-management') as management_model:
+        app_name = 'dummy-source'
+
+        deploy_local_charm(management_model, app_name)
+
+        offer_url = assert_offer_is_listed(
+            management_model, app_name, offer_name='kitchen-sink')
+        assert_offer_can_be_deleted(management_model, offer_url)
+
+        offer_url = assert_offer_is_listed(management_model, app_name)
+        assert_offer_can_be_deleted(management_model, offer_url)
+
+
+def assert_offer_is_listed(client, app_name, offer_name=None):
+    """Assert that an offered endpoint is listed.
+
+    :param client: ModelClient for model to use.
+    :param app_name: Name of the deployed application to make an offer for.
+    :param offer_name: If not None is used to name the endpoint offer.
+    :return: String URL of the resulting offered endpoint.
+    """
+    log.info('Assessing {} offers.'.format(
+        'named' if offer_name else 'unnamed'))
+
+    expected_url, offer_key = offer_endpoint(client, app_name, offer_name)
+    offer_output = yaml.safe_load(
+        client.get_juju_output('offers', '--format', 'yaml'))
+
+    fully_qualified_offer = '{controller}:{offer_url}'.format(
+        controller=client.env.controller.name,
+        offer_url=offer_output[offer_key]['offer-url'])
+    try:
+        if fully_qualified_offer != expected_url:
+            raise JujuAssertionError(
+                'Offer URL mismatch.\n{actual} != {expected}'.format(
+                    actual=offer_output[offer_key]['offer-url'],
+                    expected=expected_url))
+    except KeyError:
+        raise JujuAssertionError('No offer URL found in offers output.')
+
+    log.info('PASS: Assert offer is listed.')
+    return expected_url
+
+
+def assert_offer_can_be_deleted(client, offer_url):
+    """Assert that an offer can be successfully deleted."""
+    client.juju('remove-offer', (offer_url), include_e=False)
+    offer_output = yaml.safe_load(
+        client.get_juju_output('offers', '--format', 'yaml'))
+
+    if offer_output != {}:
+        raise JujuAssertionError(
+            'Failed to remove offer "{}"'.format(offer_url))
+    log.info('PASS: Assert offer is removed.')
+
+
+def ensure_cmr_offer_consumption(first_client, second_client):
```

Comment:
There's a semantic difference between consume and relate.
An offer may be able to be consumed but not related to. Doing a "relate" implicitly consumes first and then relates. It's also possible just to consume.

Reply:
Also, if an offer is consumed, we need to ensure that status changes are picked up etc